### PR TITLE
SearchKit - Fix `str_starts_with()` error on php74

### PR DESCRIPTION
**Pull request by totten**
Overview
----------------------------------------

Follow-up to #30007. Fix error regarding `str_starts_with()` when running on dmaster and PHP 7.4.

Before
----------------------------------------

Navigate to... basically any web-page. See the error:

```
Error message
Error: Class 'Symfony\Polyfill\Php80\Php80' not found in str_starts_with() (line 32 of /home/totten/bknix/build/dmaster/web/sites/all/modules/civicrm/vendor/symfony/polyfill-php80/bootstrap.php).
```

After
----------------------------------------

Works

Comments
----------------------------------------

SearchKit is an early-adopter of upcoming changes in civix. This particular code was previously discussed at https://github.com/totten/civix/pull/331#discussion_r1573039679. Generally, this PR just brings the early-update for SK into closer alignment with current civix drafts.

### CODE REVIEW

**Comment by totten** [comment]
Similar errors are appearing on scheduled test-runs ([example](https://test.civicrm.org/job/CiviCRM-Test-QLow/20091/console)) and test-runs for other repos ([example](https://test.civicrm.org/duderino/batch/Civix-PR/20866676)). I'm going to go ahead and merge.

**Comment by eileenmcnaughton** [comment]
@totten I'm confused - we use str_starts_with, relying on the polyfill, in loads of places - is it failing on us in other places?

**Comment by colemanw** [comment]
I'm confused too. Seems to be a missing piece of explanation in the description... I'm guessing it has something to do with boot-order and that polyfills load after civix files?

**Comment by totten** [comment]
@eileenmcnaughton I doubt `str_starts_with()` fails in most places. I'd agree with @colemanw that it has to do with boot-order.

This snippet is part of a classloader -- and that classloader is registered during bootstrap. We'd have to inspect with a debugger or debug-logging to make full picture. But generally, I'd say the classloader runs early... so it should be written conservatively...

**Comment by colemanw** [comment]
> This snippet is part of a classloader -- and that classloader is registered during bootstrap.

Ah, that would explain it.

File: ext/search_kit/info.xml [hunk 1]
```diff
@@ -28,7 +28,7 @@
     <psr0 prefix="CRM_" path="."/>
     <psr4 prefix="Civi\" path="Civi"/>
   </classloader>
-  <upgrader>CiviMix\Schema\Search\AutomaticUpgrader</upgrader>
+  <upgrader>CiviMix\Schema\SearchKit\AutomaticUpgrader</upgrader>
```
Comment (colemanw) [inline ext/search_kit/info.xml:31]: @totten why this change? The civix namespace is CRM/Search, not CRM/SearchKit...?
